File: 단계별로 풀어보기/15. 약수, 배수와 소수 2/17103.py
num = 1000000
lst = [0, 0] + [1] * (num - 1)
primeLst = []

for i in range(2, num + 1):
    if lst[i] == 1:
        primeLst.append(i)

        for j in range(2 * i, num, i):
            lst[j] = 0

primeSet = set(primeLst)
testCase = int(input())

for _ in range(testCase):
    count = 0
    n = int(input())
    for i in primeLst:
        if i >= n:
            break
        # 앞 뒤 바뀐것도 다 세어주고 마지막에 나누기 2
        if (n - i) in primeSet:
            count += 1
            # 한가지 숫자일 경우 1을 더해줌
            if (n-i) == i:
                count += 1
    print(count // 2)

File: 단계별로 풀어보기/15. 약수, 배수와 소수 2/17103.cpp
#include <iostream>
#include <algorithm>
using namespace std;

const int num = 1000000;
int arr[num];

int main()
{
	int testCase;

	fill(arr, arr + num, 1);

	for (int i = 2; i <= num; i++)
	{
		if (arr[i] == 1)
			for (int j = i * 2; j <= num; j += i)
				arr[j] = 0;
	}

	cin >> testCase;
	while (testCase--)
	{
		int n, count;
		count = 0;
		cin >> n;
		
		for (int i = 2; i * 2 <= n; i++)
			if (arr[i] && arr[n - i])
				count++;
		cout << count << endl;
	}
	return 0;
}